apps/api/src/domain/models/audio-item/entities/audio-item.entity.ts

```diff
@@ -123,6 +125,25 @@ export class AudioItem<T extends CoscradText = string> extends Resource {
         } as DeepPartial<DTO<this>>);
     }
 
+    addLineItemToTranscript(newItemDto: DTO<TranscriptItem>): ResultOrError<this> {
+        const newItem = new TranscriptItem(newItemDto);
+
+        const timeBounds = this.getTimeBounds();
+
+        const { inPoint, outPoint } = newItem;
+
+        if ([inPoint, outPoint].some((point) => !isNumberWithinRange(point, timeBounds)))
```

Comment:
To what extent should we just defer to invariant validation? In some ways, that makes this duplicated logic. But if these aren't invariant rules, we can't check existing DTOs from the db, which is helpful especially if migrations are missed in conjunction with changes to these rules.

But if we wait until the invariant errors are hit (upon calling `.safeClone({...})`, we aren't failing fast and the error message becomes more obscure).